### ISSUE TITLE
Allow Connection Limit when running Direct Mode 

### ISSUE THREAD

**Issue by edwarder86** [Azure/azure-documentdb-dotnet#428]
When running in Gateway Mode there is a connection limit setting to limit the number of open connections to CosmosDB.  This is useful when running an Azure Function that connects to CosmosDB as it keeps the function from exceeding the 300 connection limit enforced on functions running in Azure.  

However, this connection limit setting is ignored in Direct Mode.  As a result I have found that if I run Function with Direct Mode I see Connect Limit errors when large spikes occur on my Function (which is an HTTP Trigger). This has forced me to run Gateway mode rather than Direct mode.  I would much rather run Direct mode and eliminate the extra latency of Gateway mode.

Please consider enforcing the Connection Limit in  Direct Mode.  BTW, I am running the DocumentClient as a singleton.

**Comment by ealsur** [comment]
@edwarder86 You mention that you are using the DocumentClient as a singleton, is that implemented as a static instance?

**Comment by edwarder86** [comment]
Yes, it is a static instance.

**Comment by ealsur** [comment]
Thanks! Just making sure. Aren't you connecting to any other service in the Function? Any other HTTP request that could be raising the connection limit?

As far as I know, using a static / singleton instance should not raise the amount of connections.

**Comment by ealsur** [comment]
Additional info regarding the Connection Limit in Functions: https://github.com/Azure/azure-functions-host/wiki/Managing-Connections

**Comment by edwarder86** [comment]
CosmosDB is the only thing it connects to.  The problem is that I'm calling the function millions of times a day and sometimes in very large spikes.  Using Gateway Mode and setting a limit of 200 works fine for me.  If Direct Mode would support the same Conneciton Limit as Gateway Mode does I am confident that it work work just fine as well.

**Comment by ealsur** [comment]
Makes sense, will leave the comments to the SDK experts, I'm not familiar on how Direct Mode works with connections as a Singleton, just wanted to rule out any common issue on Functions environment,

**Comment by hemantkr** [comment]
We are working on adding this feature. Please stay tuned.

**Comment by edwarder86** [comment]
Thank you for the update.

**Comment by balbany** [comment]
Any update on this? We are having to resort to the same strategy (but would rather use Direct Mode for performance when possible).
